tensorflow/stream_executor/dnn.h

```diff
@@ -701,20 +704,35 @@ class AlgorithmConfig {
  public:
   AlgorithmConfig()
       : algorithm_(kDefaultAlgorithm),
-        algorithm_no_scratch_(kDefaultAlgorithm) {}
-  explicit AlgorithmConfig(AlgorithmType algorithm)
-      : algorithm_(algorithm), algorithm_no_scratch_(kDefaultAlgorithm) {}
-  AlgorithmConfig(AlgorithmType algorithm, AlgorithmType algorithm_no_scratch)
-      : algorithm_(algorithm), algorithm_no_scratch_(algorithm_no_scratch) {}
+        use_tensor_op_math_(false),
+        algorithm_no_scratch_(kDefaultAlgorithm),
+        use_tensor_op_math_no_scratch_(false) {}
+  explicit AlgorithmConfig(AlgorithmType algorithm, bool use_tensor_op_math=false)
+      : algorithm_(algorithm), use_tensor_op_math_(use_tensor_op_math),
+        algorithm_no_scratch_(kDefaultAlgorithm), use_tensor_op_math_no_scratch_(false) {}
+  AlgorithmConfig(AlgorithmType algorithm, AlgorithmType algorithm_no_scratch,
+                 bool use_tensor_op_math=false, bool use_tensor_op_math_no_scratch=false)
```

Comment:
nit: even without merge the two types into a single type, I think the order "algorithm, bool, algorithm, bool" works better.

Also I'm not a big fan of default arguments in C++. Overloaded functions tend to work better. But it's up to you.

Reply:
Types merged.